### ISSUE TITLE
Option to MONITOR a file when RELOAD is detected

### ISSUE THREAD

**Issue by PLP757** [notepad-plus-plus/notepad-plus-plus#5583]
The "Monitoring (tail -f)" feature is great.

When a file on disk changes and the RELOAD message is displayed (i.e. "This file has been modified by another program. Do you want to reload it?") the choices are YES or NO.
 Please add a third choice MONITOR

**Comment by Xiad32** [comment]
What would MONITOR do?

**Comment by PLP757** [comment]
The third choice of MONITOR would activate the "monitoring (tail -f)" feature.

"Monitoring" a file was introduced to notepad++ at least 3 years ago.
With monitoring switched on, a file will automatically be reloaded if it changes on disk.

The "monitoring" command is found at the bottom of the VIEW menu, or by clicking the "eyeball"-button on the toolbar.

Currently, if a file is constantly changing on disk the RELOAD message will be displayed over and over.
Adding the third choice MONITOR would switching on the "monitoring" function and the file would be automatically reloaded and the RELOAD message would not keep popping up.

**Comment by xylographe** [comment]
@PLP757
What about "Update silently" in  "Settings > Preferences > Miscellaneous > File Status Auto-Detection"?

**Comment by PLP757** [comment]
@xylographe
While your suggestion offers a way to constantly update files in Notepad++, that option applies to all files all the time.
This may not be what the user wants.
They may always want to be told if something is changing on disk and be made aware of it, rather than the "update silent" option.
If the user chooses the "update silent" option they will never see the "reload" message anyway.

But for users who do see the "reload" message (i.e. "This file has been modified by another program. Do you want to reload it?") then an additional choice of MONITOR adds extra convenience.
NO means "don't update"
YES means "update once now, until next time"
MONITOR means "update now and keep updating this file".

Right now, in this situation, I would click YES and then I would click the "eyeball" on the toolbar.

I was suggesting that adding a third button/choice to the "reload" message (to activate what is otherwise existing functionality) would streamline things for the user.

**Comment by Xiad32** [comment]
@PLP757 I agree. I usually change files outside the editor. Sometimes it would be useful to "MONITOR" a single file while I would not other changes to be overwritten. But if you add MONITOR to a file, one would need a way to disable it for that file too!

**Comment by PLP757** [comment]
@Xia32
Thank you for your agreement.
But I disagree with the last part... "one would need a way to disable it for that file too!"

You are talking about a radical change in functionality.
If I was editing a file in Notepad++, I would always want to know when it was changed on disk by another process.  I wouldn't want to have a "stale" copy in Notepad++.  I would edit it, unaware of other changes and then try to save it... what happens then?

All I am asking for is one extra button on the RELOAD message to activate existing functionality and to save me a mouse-click.
Me request is VERY simple.
Please don't complicate things.

**Comment by Xiad32** [comment]
@PLP757
Not trying to complicate it, but understand it. One needs to understand how a feature will be used and the implications of adding it. Not all users will use it or understand it the same way you do. 
What I meant by disable it is disabling the auto reload not disabling the prompt to reload. Another thing, the user will mentally have to remember which files are set to autoreload and which are not, unless you show it somehow. Add to that, that if someone clicks monitor by mistake, what would he do to cancel it. Moreover, will this carry over when the application is restarted or the file is closed then reopened. Its almost always much bigger than adding a button. 
I personally would use it. But don't forget that a mistake can wipe out hours of work.

**Comment by PLP757** [comment]
@Xiad32
I think you ARE missing the point and you ARE complicating it.

The Monitor function with auto-reload already exists in Notepad++
It is represented by an "eyeball" on the toolbar.
When a file is being monitored the "eyeball" icon replaces the "save-disc" icon on the edit tab for the file.

So... 
users do not have to remember which files are set to auto-reload as it is indicated on the tab
users can cancel the auto-reload at any time by pressing the "eyeball" on the toolbar or by selecting the "monitoring" from the View menu.

Currently, the monitor function does not carry over when the application is restarted
If you think that is worthwhile functionality that should be add, then start a new thread to discuss the pros and cons of that.

All I am asking for is one extra button on the RELOAD message to activate existing functionality and to save me a mouse-click.
 Me request is VERY simple.
 Please don't complicate things.

**Comment by Xiad32** [comment]
@PLP757
I see. Well, in that case your suggestion actually would make this feature more discoverable. placing that button would be useful and well placed. 👍Mind my questions. I believe that any change should be discussed from the prespective of all users.

**Comment by PLP757** [comment]
@Xiad32
That's it!  I want to make a this feature more discoverable/accessible.

As for your suggestion about monitoring files that were previously monitored, I had a similar suggestion... see issue #2187... but someone closed it down.
I even created a version of this issue before #2186, but that was closed too.

I hope with all this discussion between you and me, that someone can pick-up this simple issue and implement it.

Thank you for your comments.

**Comment by Xiad32** [comment]
I've been looking at the code. Are there any custom message boxes (with user defined buttons) used anywhere in the code? I feel that creating a custom message box with the localization might be some work. Can someone help?

**Comment by xylographe** [comment]
Perhaps, `DoSaveOrNotBox` (`IDD_DOSAVEORNOTBOX`).
